boefjes/app.py

```diff
@@ -84,28 +85,32 @@ def start_working(
         for queue in queues:
             try:
                 logger.info(f"Popping from queue {queue.id}")
-                task = scheduler_client.pop_task(queue.id)
+                p_item = scheduler_client.pop_item(queue.id)
             except (RequestHTTPError, HTTPError, ConnectionError):
                 logger.exception("Popping task from scheduler failed")
                 time.sleep(10 * settings.poll_interval)
                 continue
-            except Exception:
+            except:  # noqa
                 logger.exception("Popping task from scheduler failed")
                 time.sleep(10 * settings.poll_interval)
                 continue
 
-            if not task:
+            if not p_item:
                 logger.info(f"Queue {queue.id} empty")
                 continue
 
             try:
-                logger.info(f"Handling task[{task.data.id}]")
-                item_handler.handle(task.data)
+                logger.info(f"Handling task[{p_item.data.id}]")
+                item_handler.handle(p_item.data)
+                scheduler_client.patch_task(str(p_item.id), TaskStatus.COMPLETED)
+                logger.info(f"Set task status to completed in the scheduler for task[{p_item.data.id}]")
             except StopWorking:
                 logger.info("Stopping worker...")
                 return
             except:  # noqa
```

Comment:
This was wrong here when we just logged the exception. Now that we want to save that the task has failed, I think this one of the rare cases that a bare except should be used, but we should reraise the exception if it isn't a subclass of Exception.